plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -571,6 +573,49 @@ public void testTopNPushdown()
                 .isFullyPushedDown();
     }
 
+    @Test
+    public void testCaseSensitiveTopNPushdown()
+    {
+        try (TestTable testTable = new TestTable(
+                postgreSqlServer::execute,
+                "tpch.test_case_sensitive_topn_pushdown",
+                "(a_string varchar(10), a_char char(10), a_bigint bigint)",
+                List.of(
+                        "'A', 'A', 1",
+                        "'B', 'B', 2",
+                        "'a', 'a', 3",
+                        "'b', 'b', 4"))) {
+            assertThat(query("SELECT a_bigint FROM " + testTable.getName() + " ORDER BY a_string ASC LIMIT 2"))
+                    .ordered()
+                    .isNotFullyPushedDown(TopNNode.class);
```

Comment:
Do we want to assert if pushdown happened or only if the results are correct (because some databases would return correct results even with pushdown).

Or we can revisit this when we try and extract this to BaseJdbcConnectorTest.

Reply:
> Or we can revisit this when we try and extract this to BaseJdbcConnectorTest.

this